lib/ansible/plugins/connection/vmware_tools.py

```diff
@@ -0,0 +1,369 @@
+# Copyright (c) 2018 Deric Crago <deric.crago@gmail.com>
+# Copyright (c) 2018 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+import re
+from os.path import dirname, exists, getsize
+from socket import gaierror
+from ssl import SSLEOFError, SSLError
+from time import sleep
+
+import requests
+import urllib3
+
+from ansible.errors import AnsibleError, AnsibleFileNotFound
+from ansible.module_utils._text import to_bytes, to_native
+from ansible.plugins.connection import ConnectionBase
+from ansible.utils.path import makedirs_safe
+
+try:
+    from pyVim.connect import Disconnect, SmartConnect, SmartConnectNoSSL
+    from pyVmomi import vim
+
+    HAS_PYVMOMI = True
+except ImportError as e:
+    HAS_PYVMOMI = False
+    PYVMOMI_IMPORT_ERROR = e
+
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+    author: Deric Crago <deric.crago@gmail.com>
+    connection: vmware_tools
+    short_description: Execute modules via VMware Tools.
+    description:
+      - Execute modules via VMware Tools.
+      - "Note: Windows VMs will need to have C(ansible_shell_type: powershell) set."
+    version_added: "2.8"
+    requirements:
+      - pyvmomi (python library)
+    options:
+      connection_address:
+        description:
+          - Address for the connection (vCenter or ESXi Host).
+        vars:
+          - name: ansible_vmware_tools_connection_address
+        required: True
+      connection_username:
+        description:
+          - Username for the connection.
+          - "Requires the following permissions on the VM:
+               - VirtualMachine.GuestOperations.Execute
+               - VirtualMachine.GuestOperations.Modify
+               - VirtualMachine.GuestOperations.Query"
+        vars:
+          - name: ansible_vmware_tools_connection_username
+        required: True
+      connection_password:
+        description:
+          - Password for the connection.
+        vars:
+          - name: ansible_vmware_tools_connection_password
+        required: True
+      connection_verify_ssl:
+        description:
+          - Verify SSL for the connection.
+          - "Note: This will verify SSL for both C(connection_address) and the ESXi host running the VM."
+        vars:
+          - name: ansible_vmware_tools_connection_verify_ssl
```

Comment:
make it more like get_url
```suggestion
          - name: ansible_vmware_tools_validate_certs
```